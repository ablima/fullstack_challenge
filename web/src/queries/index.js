import gql from "graphql-tag";

export const GET_PRODUCTS = gql`
  query Products($id: Int, $name: String, $categoryId: Int){
    getProducts(id: $id, name: $name, categoryId: $categoryId){
      id
      name
      description
      qnt
      price
      thumbnails
    }
  }
`

export const ADD_PRODUCT_TO_CART = gql`
  mutation Product($id: Int){
    addProductToCart(id: $id){
      id
    }
  }
`;

export const REMOVE_PRODUCT_FROM_CART = gql`
  mutation Product($id: Int){
    removeProductFromCart(id: $id){
      id
    }
  }
`;

export const GET_CATEGORIES = gql`
  query {
    getCategories{
      id
      name
    }
  }
`;

export const CREATE_ORDER = gql`  
  mutation Order($order: NewOrder){
    createOrder(order: $order){
      id
    }
  }
`;