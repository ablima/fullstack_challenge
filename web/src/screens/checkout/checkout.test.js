import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import { MockedProvider } from '@apollo/client/testing';
import { CREATE_ORDER } from '../../queries';
import CheckoutScreen from './index';

const mocks = [
  {
    request: {
      query: CREATE_ORDER,
      variables: {
        order: {
          cardNumber: "0000111122223333",
          userId: 1,
          products: [
            {
              id: 17,
              qnt: 1
            },
            {
              id:18,
              qnt: 3
            }
          ]
        }
      }
    },
    result: {}
  }
];

test('Check if render is correct', () => {
  const component = renderer.create(
    <MockedProvider mocks={mocks} addTypename={false}>
      <MemoryRouter initialEntries={["/checkout"]}>
        <CheckoutScreen />
      </MemoryRouter>
    </MockedProvider>    
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});