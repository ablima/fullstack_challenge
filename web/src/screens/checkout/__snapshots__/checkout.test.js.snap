// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if render is correct 1`] = `
<div
  id="outer-container"
>
  <div
    className="header"
    id="header"
  >
    <img
      className="logo"
      onClick={[Function]}
      src="storeLogo.png"
    />
    <div
      className="cartIcon"
    >
      <img
        className="icon"
        onClick={[Function]}
        src="cart.svg"
      />
      <div
        className="animated"
        style={
          Object {
            "animationDelay": "undefinedms",
            "animationDuration": "undefinedms",
            "opacity": 0,
            "pointerEvents": "none",
          }
        }
      >
        <div
          className="cartContainer"
        >
          <span>
            Sem produtos no carrinho
          </span>
        </div>
      </div>
    </div>
  </div>
  <main
    id="page-wrap"
  >
    <div
      className="mainContent"
    >
      <div
        className="categoriesContainer"
      />
      <div
        className="root"
      >
        <div
          className="infoContainer"
        >
          <h1>
            Carrinho
          </h1>
        </div>
        <div
          className="checkoutContainer"
        >
          <h3>
            Total: 
            R$Â 0,00
          </h3>
          <button
            onClick={[Function]}
          >
            Comprar
          </button>
        </div>
      </div>
    </div>
  </main>
</div>
`;
